substrate/client/network/sync/src/strategy.rs

```diff
@@ -92,15 +103,68 @@ pub enum SyncingAction<B: BlockT> {
 		number: NumberFor<B>,
 		justifications: Justifications,
 	},
-	/// Syncing strategy has finished.
+	/// Strategy finished. Nothing to do, this is handled by `SyncingStrategy`.
```

Comment:
Having `SyncingAction::Finished` is needed to be able to convert specific actions using `Into::into()` here: https://github.com/paritytech/polkadot-sdk/blob/0d245f1fdecb52e3547fe63558622ea99b9af2d7/substrate/client/network/sync/src/strategy.rs#L483-L488

Otherwise we'd need to introduce match statements for every startegy in [`SyncingStartegy::actions()`](https://github.com/paritytech/polkadot-sdk/blob/0d245f1fdecb52e3547fe63558622ea99b9af2d7/substrate/client/network/sync/src/strategy.rs#L480).